uvicorn/config.py

```diff
@@ -444,6 +444,13 @@ def load(self) -> None:
 
         self.lifespan_class = import_from_string(LIFESPAN[self.lifespan])
 
+        if (self.reload or self.workers > 1) and not isinstance(self.app, str):
```

Comment:
Shouldn't this check go in the constructor? By the time we get here, we are already in the subprocess I believe...

Reply:
I think this check should be combined with `import_from_string(self.app)`.

Reply:
Why? Isn't it preferable to "fail fast" in the main process rather than failing later in possibly N subprocesses?

Reply:
When the user provides a string, checks the type in one place, but tries to import it in another far away, don't you think this is unreasonable?

If it is for this purpose, then `import_from_string(self.app)` should also "fail fast".